WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantActivity.kt

```diff
@@ -38,22 +55,45 @@ class AuthDeviceGrantActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_auth)
-        val viewModel by viewModels<AuthDeviceGrantViewModel>()
+        setContent { AuthenticateScreen(deviceGrantViewModel = viewModel()) }
+    }
+}
 
-        // Start the OAuth flow when the user presses the button
-        findViewById<View>(R.id.authenticateButton).setOnClickListener {
-            viewModel.startAuthFlow()
-        }
+@Composable
+fun AuthenticateScreen(deviceGrantViewModel: AuthDeviceGrantViewModel) {
+    val listState = rememberScalingLazyListState()
+    val uiState = deviceGrantViewModel.uiState.collectAsState()
+    val localContext = LocalContext.current
 
-        // Show current status on the screen
-        viewModel.status.observe(this) { statusText ->
-            findViewById<TextView>(R.id.status_text_view).text = resources.getText(statusText)
+    Scaffold(
+        timeText = { TimeText(modifier = Modifier.scrollAway(listState)) },
+        vignette = {
+            Vignette(vignettePosition = VignettePosition.TopAndBottom)
+        },
+        positionIndicator = {
+            PositionIndicator(scalingLazyListState = listState)
         }
-
-        // Show dynamic content on the screen
-        viewModel.result.observe(this) { resultText ->
-            findViewById<TextView>(R.id.result_text_view).text = resultText
+    ) {
+        ScalingLazyColumn(
```

Comment:
Here you can use ScalingLazyColumn from Horologist, see https://github.com/android/wear-os-samples/blob/main/ComposeStarter/app/src/main/java/com/example/android/wearable/composestarter/presentation/MainActivity.kt#L156 and https://github.com/android/wear-os-samples/blob/main/ComposeStarter/app/src/main/java/com/example/android/wearable/composestarter/presentation/MainActivity.kt#L143